### ISSUE TITLE
Escape for `pulumi config set` on Windows CMD and PowerShell

### ISSUE THREAD

**Issue by guitarrapc** [pulumi/pulumi#3594]
## TL;DR

`pulumi config set data '{"active": true, "nums": [1,2,3]}'` works on bash, but not on CMD and PowerShell.

CMD and PowerShell requires additional escape.

```
pulumi config set data "{""active"""": true, ""nums"""": [1,2,3]}"
```

## Actual behavior

Document told `pulumi config set data '{"active": true, "nums": [1,2,3]}'` to set JSON to the config.

> REF: https://www.pulumi.com/docs/intro/concepts/programming-model/#config

Running on Bash will set valid JSON to the config.

```shell
$ pulumi config set data '{"active": true, "nums": [1,2,3]}'
$ pulumi config
# result with bash
data         {"active": true, "nums": [1,2,3]}
```

However CMD will result error as follows.

```shell
CMD> pulumi config set data '{"active": true, "nums": [1,2,3]}'

Configuration values can be accessed when a stack is being deployed and used to configure behavior.
If a value is not present on the command line, pulumi will prompt for the value. Multi-line values
may be set by piping a file to standard in.

The `--path` flag can be used to set a value inside a map or list:

    - `pulumi config set --path outer.inner value` will set the value of `outer` to a map `inner: value`.
    - `pulumi config set --path names[0] a` will set the value to a list with the first item `a`.

Usage:
  pulumi config set <key> [value] [flags]
```

PowerShell will not return error, but register invalid JSON, as `key` missing double quote.

```
PS> pulumi config set data '{"active": true, "nums": [1,2,3]}'
PS> pulumi config
data         {active: true, nums: [1,2,3]}
```

## Workaround

You need escape JSON key with additional double quote for both CMD and PowerShell.

* begin: 2 double quote `""`
* end: 4 double quote `""""`

> PowerShell's JSON string should be quote with `'` for simplify problem. Double quote requires additional escape:(

```shell
CMD> pulumi config set data "{""active"""": true, ""nums"""": [1,2,3]}"
CMD> pulumi config
data         {"active": true, "nums": [1,2,3]}
```

```shell
PS> pulumi config set data '{""active"""": true, ""nums"""": [1,2,3]}'
PS> pulumi config
data         {"active": true, "nums": [1,2,3]}
```

## Expected behavior

This may not a Pulumi problem but CMD and PowerShell shell design.
Additional note to a doc would be appreciated.

PowerShell argument would be fixable...?
